pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -0,0 +1,353 @@
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Infof("CnsFileAccessConfig resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		log.Errorf("Error reading the CnsFileAccessConfig with name: %q on namespace: %q. Err: %+v",
+			request.Name, request.Namespace, err)
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Initialize backOffDuration for the instance, if required.
+	backOffDurationMapMutex.Lock()
+	var timeout time.Duration
+	if _, exists := backOffDuration[instance.Name]; !exists {
+		backOffDuration[instance.Name] = time.Second
+	}
+	timeout = backOffDuration[instance.Name]
+	backOffDurationMapMutex.Unlock()
+
+	if instance.DeletionTimestamp != nil {
+		// TODO:
+		// Identify the network provider (NSX-T or VDS).
+		// Find external facing IP of the VM depending on NSXT or VDS - Analysis on how to expose external facing IP of TKG and Pod VMs.
+		// Get ExternalIPtoClientVms[IP] list from FileVolumeClients instance for SV-PVC.
+		// If ExternalIPtoClientVms[IP] exists and CnsFileAccessConfig.Spec.VirtualMachine.Name is the only value in the ExternalIPtoClientVms[IP], then
+		//   Call CNS API to remove the IPAddresses on file volume.
+		// If not,
+		//   Do nothing (as there are other virtual machines on same external IP using this volume)
+		// Call RemoveClientVmFromIPList() to remove req.NodeID to list of nodes exposed by IP referencing this file volume.
+		removeFinalizerFromCRDInstance(ctx, instance)
+		return reconcile.Result{}, nil
+	}
+
+	// If the CnsFileAccessConfig instance is already successful,
+	// and not deleted by the user, remove the instance from the queue.
+	if instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Cleanup instance entry from backOffDuration map
+		backOffDurationMapMutex.Lock()
+		delete(backOffDuration, instance.Name)
+		backOffDurationMapMutex.Unlock()
+		return reconcile.Result{}, nil
+	}
+
+	log.Infof("Reconciling CnsFileAccessConfig with instance: %q from namespace: %q. timeout %q seconds", instance.Name, instance.Namespace, timeout)
+	if !instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Get the virtualmachine instance
+		vm, err := getVirtualMachine(ctx, r.vmOperatorClient, instance.Spec.VMName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get virtualmachine instance for the VM with name: %q. Error: %+v", instance.Spec.VMName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Found virtualMachine instance for VM with name: %q: +%v", instance.Spec.VMName, vm)
+
+		// Set ownerRef on CnsFileAccessConfig instance (in-memory) to VM instance
+		setInstanceOwnerRef(instance, instance.Spec.VMName, vm.UID)
+
+		// Add finalizer.
+		instance.Finalizers = append(instance.Finalizers, cnsoperatortypes.CNSFinalizer)
```

Comment:
CNS Operator finalizer only needs to be added if it doesnt exist right?

Reply:
This was there originally, I think I removed while addressing some lint checks. Added back the finalizer check.